src/compiler/checker.ts

```diff
@@ -5860,6 +5888,70 @@ namespace ts {
             return links.resolvedType;
         }
 
+        /**
+         * Since the source of spread types are object literals, which are not binary,
+         * this function should be called in a left folding style, with left = previous result of getSpreadType
+         * and right = the new element to be spread.
+         */
+        function getSpreadType(left: Type, right: Type, symbol: Symbol): ResolvedType | IntrinsicType {

```

Comment:
The `symbol` parameter doesn't make sense. This should just create an anonymous type with no symbol.